cmd/print_test.go

```diff
@@ -0,0 +1,39 @@
+package cmd
+
+import (
+	"k8s.io/cli-runtime/pkg/printers"
+	"reflect"
+	"testing"
+)
+
+func Test_getResourcePrinter(t *testing.T) {
+	testCases := []struct {
+		outputFormat    string
+		expectedPrinter printers.ResourcePrinter
+	}{
+		{
+			outputFormat:    "json",
+			expectedPrinter: &printers.JSONPrinter{},
+		},
+		{
+			outputFormat:    "yaml",
+			expectedPrinter: &printers.YAMLPrinter{},
+		},
+		{
+			outputFormat:    "",
+			expectedPrinter: &printers.YAMLPrinter{},
+		},
+		{
+			outputFormat:    "invalid",
+			expectedPrinter: nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		result := getResourcePrinter(tc.outputFormat)
+
+		if !reflect.DeepEqual(result, tc.expectedPrinter) {
```

Comment:
I think for testing purposes `if diff := cmp.diff(....)` is a slightly better suite option. It also more common to use in unitests. 

[Here](https://pkg.go.dev/github.com/google/go-cmp/cmp#example-Diff-Testing) is one example

Reply:
I think using reflect.DeepEqual is good enough for just checking type comparison, while avoiding the extra dependency